### PR TITLE
collapsible_if: split collapsible_else_if into its own lint so we can enable/disable it particularly

### PR DESCRIPTION
**Pull request by matthiaskrgr**
This splits up clippy::collapsible_if into collapsible_if for
if x {
  if y { }
}
=>
if x && y { }

and collapsible_else_if for

if x {
} else {
 if y { }
}

=>
if x {

} else if y {

}

so that we can lint for only the latter but not the first if we desire.

changelog: collapsible_if: split up linting for if x {} else { if y {} } into collapsible_else_if lint

### CODE REVIEW

File: clippy_lints/src/collapsible_if.rs [hunk 1]
```diff
@@ -81,6 +98,14 @@ impl EarlyLintPass for CollapsibleIf {
     }
 }
 
+impl EarlyLintPass for CollapsibleIfElse {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &ast::Expr) {
+        if !expr.span.from_expansion() {
+            check_if(cx, expr)
+        }
+    }
```
Comment (flip1995) [inline clippy_lints/src/collapsible_if.rs:106]: Why implement a new lint pass for this. Allowing lints is only dependent on the lint name, not the lint pass.

File: clippy_lints/src/collapsible_if.rs [hunk 2]
```diff
@@ -54,24 +76,19 @@ declare_clippy_lint! {
...
-    ///
     /// if x {
     ///     …
     /// } else if y {
     ///     …
     /// }
     /// ```
-    pub COLLAPSIBLE_IF,
+    pub COLLAPSIBLE_ELSE_IF,
     style,
```
Comment (flip1995) [inline clippy_lints/src/collapsible_if.rs:86]: Having the split out lint in the `style` category will result in people having to allow this lint again in their code, if they don't like it. 

I guess the argument that this is fine, is that this case of the lint is easier to fix than the other case?

Comment (matthiaskrgr) [reply clippy_lints/src/collapsible_if.rs:86]: Yeah, hopefully it won't be too annoying. :/
I also think that the  `if {} else { if {} } ` warning is actually more useful than the `if { if {} }` warning.

Comment (flip1995) [reply clippy_lints/src/collapsible_if.rs:86]: Looking at your Rust PR, where you linked to this, it seems Joshua also agrees. So I'm ok with this.

File: clippy_lints/src/collapsible_if.rs [hunk 3]
```diff
@@ -54,21 +76,15 @@ declare_clippy_lint! {
...
-    "`if`s that can be collapsed (e.g., `if x { if y { ... } }` and `else { if x { ... } }`)"
+    "nested `else`-`if` expressions that can be collapsed (e.g., `else { if x { ... } }`)"
 }
 
 declare_lint_pass!(CollapsibleIf => [COLLAPSIBLE_IF]);
```
Comment (flip1995) [inline clippy_lints/src/collapsible_if.rs:90]:
```suggestion
declare_lint_pass!(CollapsibleIf => [COLLAPSIBLE_IF, COLLAPSIBLE_ELSE_IF]);
```

**Comment by flip1995** [comment]
r=me with CI passing.

**Pull request merged**